commit 9f20263f224cf85a81b16fe1570439cecf5ef48b
Author: mjw <mjw@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Wed Sep 23 11:51:47 2015 +0000

    Bug 353083 arm64 doesn't implement various xattr system calls.
    
    This enables setxattr, lsetxattr, fsetxattr, fgetxattr, removexattr,
    lremovexattr, fremovexattr, listxattr, llistxattr and flistxattr on
    arm64-linux.
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@15675 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/coregrind/m_syswrap/syswrap-arm64-linux.c b/coregrind/m_syswrap/syswrap-arm64-linux.c
index 6661d1d..6cf3ba5 100644
--- a/coregrind/m_syswrap/syswrap-arm64-linux.c
+++ b/coregrind/m_syswrap/syswrap-arm64-linux.c
@@ -855,8 +855,18 @@ PRE(sys_rt_sigreturn)
 // (unknown).
 
 static SyscallTableEntry syscall_main_table[] = {
+   LINX_(__NR_setxattr,          sys_setxattr),          // 5
+   LINX_(__NR_lsetxattr,         sys_lsetxattr),         // 6
+   LINX_(__NR_fsetxattr,         sys_fsetxattr),         // 7
    LINXY(__NR_getxattr,          sys_getxattr),          // 8
    LINXY(__NR_lgetxattr,         sys_lgetxattr),         // 9
+   LINXY(__NR_fgetxattr,         sys_fgetxattr),         // 10
+   LINXY(__NR_listxattr,         sys_listxattr),         // 11
+   LINXY(__NR_llistxattr,        sys_llistxattr),        // 12
+   LINXY(__NR_flistxattr,        sys_flistxattr),        // 13
+   LINX_(__NR_removexattr,       sys_removexattr),       // 14
+   LINX_(__NR_lremovexattr,      sys_lremovexattr),      // 15
+   LINX_(__NR_fremovexattr,      sys_fremovexattr),      // 16
    GENXY(__NR_getcwd,            sys_getcwd),            // 17
    LINXY(__NR_eventfd2,          sys_eventfd2),          // 19
    LINXY(__NR_epoll_create1,     sys_epoll_create1),     // 20
@@ -1267,18 +1277,6 @@ static SyscallTableEntry syscall_main_table[] = {
 //ZZ //   GENX_(222,                    sys_ni_syscall),     // 222
 //ZZ //   PLAXY(223,                    sys_syscall223),     // 223 // sys_bproc?
 //ZZ 
-//ZZ    LINX_(__NR_setxattr,          sys_setxattr),       // 226
-//ZZ    LINX_(__NR_lsetxattr,         sys_lsetxattr),      // 227
-//ZZ    LINX_(__NR_fsetxattr,         sys_fsetxattr),      // 228
-//ZZ 
-//ZZ    LINXY(__NR_fgetxattr,         sys_fgetxattr),      // 231
-//ZZ    LINXY(__NR_listxattr,         sys_listxattr),      // 232
-//ZZ    LINXY(__NR_llistxattr,        sys_llistxattr),     // 233
-//ZZ    LINXY(__NR_flistxattr,        sys_flistxattr),     // 234
-//ZZ 
-//ZZ    LINX_(__NR_removexattr,       sys_removexattr),    // 235
-//ZZ    LINX_(__NR_lremovexattr,      sys_lremovexattr),   // 236
-//ZZ    LINX_(__NR_fremovexattr,      sys_fremovexattr),   // 237
 //ZZ    LINXY(__NR_tkill,             sys_tkill),          // 238 */Linux
 //ZZ    LINXY(__NR_sendfile64,        sys_sendfile64),     // 239
 //ZZ 
